### PR TITLE
[APP-372] Feature/delete user

### PR DESCRIPTION
**Pull request by guillaumearm**
My actual `.env.development` and `.env.production` file look like:
```
# Prismic
API_KEY=[KEY]

# Auth0
AUTH0_DOMAIN=station.eu.auth0.com
AUTH0_CLIENT_ID=[ID]

# GraphQL
GRAPHQL_URI=http://127.0.0.1:4001/graphql

```

## Test instructions
first, you need to start station api locally on branch `feature/delete-user`
please see this PR for more infos: https://github.com/getstation/api/pull/42

#### development
- yarn develop

#### bundled app
- cp .env.development .env.production
- yarn build
- yarn serve

### CODE REVIEW

File: src/pages/delete-account.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        background={DATA.background.url}
+      >
+        <DeleteAccountButton method={deleteAccount} >
+          {DATA.button_confirm_text}
+        </DeleteAccountButton>
+      </SectionFullPage>
+    </AppMinimal>
+  );
+};
+export default DownloadNextPage;
```
Comment (Nacryn) [inline src/pages/delete-account.js]: Typo in the naming of this new page (+ line 13 above)

File: package.json [hunk 2]
```diff
@@ -43,10 +52,12 @@
   ],
   "license": "MIT",
   "scripts": {
+    "clean": "gatsby clean && rm -rf storybook-static",
     "build": "gatsby build",
+    "serve": "gatsby serve",
     "develop": "gatsby develop",
     "format": "prettier --write '**/*.js'",
-    "test": "echo \"Error: no test specified\" && exit 1",
+    "test": "yarn clean && yarn build && yarn build-storybook && echo build ok",
```
Comment (alexstrat) [inline package.json]: 👍

File: src/pages/logout.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { logout } from '../utils/auth';
+
+const Logout = () => {
+  logout();
```
Comment (alexstrat) [inline src/pages/logout.js]: No imperative in render function

File: src/pages/delete-account-auth0-callback.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { handleAuthentication } from '../utils/auth';
+
+const Callback = () => {
+  handleAuthentication();
```
Comment (alexstrat) [inline src/pages/delete-account-auth0-callback.js]: No imperative in render function

File: src/pages/delete-account.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+  mutation {
+    deleteMyData {
+      success
+    }
+  }
+`;
+
+const DeleteAccountPage = props => {
+  if (!isAuthenticated()) {
+    login();
```
Comment (alexstrat) [inline src/pages/delete-account.js]: No imperative in render function

File: src/pages/delete-account.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+const DELETE_ACCOUNT = gql`
+  mutation {
+    deleteMyData {
+      success
+    }
+  }
+`;
+
+const DeleteAccountPage = props => {
```
Comment (alexstrat) [inline src/pages/delete-account.js]: This component might be better readable if it was defined as Class.

File: src/pages/delete-account.js [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    login();
+    return <p>Redirecting to login...</p>
+  }
+
+  const [deleting, setDeleting] = useState(false);
+  const [done, setDone] = useState(false);
+  const [deleteAccountError, setDeleteAccountError] = useState(null);
+
+  const DATA = props.data.content.data;
+  const user = getProfile();
```
Comment (alexstrat) [inline src/pages/delete-account.js]: ⚠️ Use of render-data outside of state and props

File: src/pages/delete-account.js [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+const DELETE_ACCOUNT = gql`
+  mutation {
+    deleteMyData {
+      success
+    }
+  }
+`;
+
+const DeleteAccountPage = props => {
+  if (!isAuthenticated()) {
```
Comment (alexstrat) [inline src/pages/delete-account.js]: ⚠️ Use of render-data outside of state and props

**Pull request merged**